core/src/main/java/org/mule/runtime/core/internal/processor/strategy/DefaultReactorSinkProvider.java

```diff
@@ -11,113 +11,63 @@
 import static java.lang.System.getProperty;
 import static java.lang.Thread.currentThread;
 import static java.lang.Thread.yield;
-import static java.util.concurrent.TimeUnit.MINUTES;
 import static org.mule.runtime.api.util.MuleSystemProperties.MULE_LIFECYCLE_FAIL_ON_FIRST_DISPOSE_ERROR;
 
-import org.mule.runtime.api.lifecycle.Disposable;
-import org.mule.runtime.core.api.construct.BackPressureReason;
 import org.mule.runtime.core.api.construct.FlowConstruct;
-import org.mule.runtime.core.api.event.CoreEvent;
 import org.mule.runtime.core.api.processor.ReactiveProcessor;
-import org.mule.runtime.core.api.processor.Sink;
-import org.mule.runtime.core.api.transaction.Transaction;
-import org.mule.runtime.core.api.transaction.TransactionCoordination;
 import org.mule.runtime.core.internal.rx.FluxSinkRecorder;
+import org.mule.runtime.core.api.event.CoreEvent;
 
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Consumer;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
-import com.github.benmanes.caffeine.cache.Cache;
-import com.github.benmanes.caffeine.cache.Caffeine;
-import com.github.benmanes.caffeine.cache.RemovalListener;
-
-import reactor.core.publisher.Flux;
 import reactor.core.publisher.FluxSink;
 
-/**
- * {@link Sink} implementation that uses a {@link Flux} for each thread that dispatches events to it.
- */
-public class StreamPerThreadSink implements Sink, Disposable {
+public class DefaultReactorSinkProvider extends AbstractReactorSinkProvider {
```

Comment:
I would add a java doc and change the name to DefaultCachedThreadReactorSinkProvider